checkbox to switch

mateosalta opened the pull request:
Change single checkbox to switch to match other control interactions, hide slider when can not be used.
Fix #128

mateosalta committed:
checkbox to switch

Change single checkbox to switch to match other control interactions, hide slider when can not be used.

cibersheep commented:
Cool!

UniversalSuperBox reviewed (changes requested):
With this change, the brightness slider disappears when "Adjust automatically" is set. However, the slider is above the switch. This causes it to look like the switch has moved when it is checked, which is quite jarring.

![Brightness & Display with automatic adjustment unchecked](https://user-images.githubusercontent.com/21966173/72736541-8527ca80-3b63-11ea-8fe7-569c4d87aa46.png)

![Brightness & Display with automatic adjustment checked](https://user-images.githubusercontent.com/21966173/72736667-c324ee80-3b63-11ea-8f30-d1635c458698.png)

We could either not hide the brightness slider or move the "Adjust automatically" switch above the slider to reduce this jarring effect.

mateosalta commented:
cool, I think I will switch them.

Disabling the control is also confusing as there is no visual, and I was trying to animate the change, but that didn't make sense either.

mateosalta committed:
Update PageComponent.qml

mateosalta committed:
Update PageComponent.qml

move display brightness back to above slider, hide on auto brightness as well

mateosalta commented:
hm, I'm thinking maybe I will move them back so they match the indicator pulldown, and revisit the bug for being able to move the slider while this is checked.

mateosalta committed:
revert hiding slider and placment of itmes

Going with matching the indicator panel, and not hiding the brightness slider (one bug at a time :))

mateosalta committed:
remove :

mateosalta commented:
Ok, I think this is at a good point now.

Changes:
- checkbox -> switch
- use "SettingsItemTitle" for title headers matching the rest of system settings
- remove double line separator in favor of a "Display" header
- keep order as orginal

UniversalSuperBox commented on plugins/brightness/PageComponent.qml:
```diff
@@ -79,12 +79,12 @@ ItemPage {
-                showDivider: false
             }
-
+            
```
The only thing changed on this line is the addition of space characters. Please remove them.

The same thing occurs on line 113.

mateosalta replied:
Cool, I think I got those back to normal :)

mateosalta committed:
Update PageComponent.qml

mateosalta committed:
Update PageComponent.qml

UniversalSuperBox commented on plugins/brightness/PageComponent.qml:
```diff
@@ -79,12 +79,11 @@ ItemPage {
-            ListItem.Standard {
+            
+            //SettingsItemTitle to be consistent with the rest of settings
+            SettingsItemTitle {
```
If this is consistent with the rest of the codebase, there is no need for a comment here.

mateosalta replied:
done, also realized I needed to remover the dividers for the titles as all the others do the same

mateosalta committed:
remove extra comments, remove dividers from titles

other system titles have no dividers shown

UniversalSuperBox commented on plugins/brightness/PageComponent.qml:
```diff
@@ -79,12 +79,11 @@ ItemPage {
             anchors.right: parent.right
-
-            ListItem.Standard {
+            
```
This line only changes whitespace, the whitespace can be removed

mateosalta replied:
hm, maybe a stray space? it removed and added the same. oh, I seee - habit from using python ;)

mateosalta committed:
Update PageComponent.qml

mateosalta committed:
Update PageComponent.qml

mateosalta committed:
Update PageComponent.qml

mateosalta reviewed (commented):
Got it :+1:

mateosalta committed:
has no divider...

forgot that this didn't have a divider not to show
